components/kyma-environment-broker/internal/runtime/provider.go

```diff
@@ -24,11 +24,17 @@ const (
 type ComponentsListProvider struct {
 	kymaVersion                      string
 	managedRuntimeComponentsYAMLPath string
+	httpClient                       HTTPDoer
+}
+
+type HTTPDoer interface {
+	Do(req *http.Request) (*http.Response, error)
 }
 
 // NewComponentsListProvider returns new instance of the ComponentsListProvider
-func NewComponentsListProvider(kymaVersion string, managedRuntimeComponentsYAMLPath string) *ComponentsListProvider {
+func NewComponentsListProvider(client HTTPDoer, kymaVersion string, managedRuntimeComponentsYAMLPath string) *ComponentsListProvider {
```

Comment:
if there is really no way to do it without injecting http doer please create separate contructor for tests

Reply:
I've added dedicated setter available only for tests